### PR TITLE
[physfs] enable static builds, use external zlib

### PR DESCRIPTION
**Pull request by codicodi**

### CODE REVIEW

File: ports/physfs/portfile.cmake [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+    URLS "https://icculus.org/physfs/downloads/physfs-${PHYSFS_VERSION}.tar.bz2"
+    FILENAME "physfs-${PHYSFS_VERSION}.tar.bz2"
     SHA512 47eff0c81b8dc3bb526766b0a8ad2437d2951867880116d6e6e8f2ec1490e263541fb741867fed6517cc3fa8a9c5651b36e3e02a499f19cfdc5c7261c9707e80
 )
 vcpkg_extract_source_archive(${ARCHIVE})
 
+vcpkg_apply_patches(SOURCE_PATH ${SOURCE_PATH}
+    PATCHES ${CMAKE_CURRENT_LIST_DIR}/export-symbols-in-shared-build-only.patch)
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" PHYSFS_STATIC)
```
Comment (ras0219-msft) [inline ports/physfs/portfile.cmake]: Ooo, this is quite cool. We should put it in some docs somewhere!

**Comment by ras0219-msft** [comment]
Looks great!

One question: does it not currently use `dllimport` and thus why a post-build header patch isn't needed?

**Comment by codicodi** [comment]
Currently it does not use `dllimport` which seems to work fine, as the library does not expose any global variables.

**Comment by alexkaratarakis** [comment]
Thanks!

**Pull request merged**